tools/analyzer_plugin/test/integration/assists/add_props_test.dart

```diff
@@ -0,0 +1,67 @@
+import 'dart:async';
+
+import 'package:over_react_analyzer_plugin/src/assist/add_props.dart';
+import 'package:over_react_analyzer_plugin/src/async_plugin_apis/assist.dart';
+import 'package:test/test.dart';
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../test_bases/assist_test_base.dart';
+
+void main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(AddPropsAssistTest);
+  });
+}
+
+@reflectiveTest
+class AddPropsAssistTest extends AssistTestBase {
+  static int expectedPriority = AddPropsAssistContributor.addPropsKind.priority;
+
+  @override
+  AsyncAssistContributor getContributorUnderTest() =>
+      AddPropsAssistContributor();
+
+  Future<void> test_noPropsAndNoParens() async {
+    var source = newSource('test.dart', '''
+import 'package:over_react/over_react.dart';
+var foo = Dom.div()('hello');
+''');
+    var selection = createSelection(source, '#Dom.div#');
+    final assists = await getAssists(selection);
```

Comment:
Do `createSelection` and `getAssists` support a selection of length 0? e.g., could you do the following?
```dart
    var selection = createSelection(source, '##Dom.div');
    final assists = await getAssists(selection);
```

Reply:
They do! I'll add some additional test cases and docs to clarify this.